torchvision/datasets/cifar.py

```diff
@@ -94,6 +94,10 @@ def __init__(self, root, train=True,
         self.data = np.vstack(self.data).reshape(-1, 3, 32, 32)
         self.data = self.data.transpose((0, 2, 3, 1))  # convert to HWC
 
+        # doing this so that it is consistent with all other datasets
+        # to return a PIL Image
+        self.data = [Image.fromarray(image) for image in self.data]
```

Comment:
This is a breaking change if users accessed the internal field `.data` before.

I wonder if we should instead add a deprecation warning in this case?

Reply:
Should `data` be an internal field? I would argue in favour of this, but I wasn't sure if that is the case fore everyone.

Reply:
Even if we do not encourage users to query it, they still do it, see for example https://github.com/pytorch/vision/pull/578#issuecomment-460615581

Reply:
We should take this up in #1080.